### PR TITLE
Ability to specify the storage location of a reference type.

### PR DESCRIPTION
**Pull request by chriseth**

### CODE REVIEW

File: libsolidity/AST.h [hunk 1]
```diff
@@ -521,6 +529,7 @@ class VariableDeclaration: public Declaration
 	bool m_isStateVariable;             ///< Whether or not this is a contract state variable
 	bool m_isIndexed;                   ///< Whether this is an indexed variable (used by events).
 	bool m_isConstant;                  ///< Whether the variable is a compile-time constant.
+	Location m_location; ///< Location of the variable if it is of reference type.
```
Comment (LianaHus) [inline libsolidity/AST.h]: would be nicer to align other comments to

File: libsolidity/Parser.cpp [hunk 2]
```diff
@@ -371,7 +396,7 @@ ASTPointer<VariableDeclaration> Parser::parseVariableDeclaration(
 	}
 	return nodeFactory.createNode<VariableDeclaration>(type, identifier, value,
 									visibility, _options.isStateVariable,
-									isIndexed, isDeclaredConst);
+									isIndexed, isDeclaredConst, location);
```
Comment (LianaHus) [inline libsolidity/Parser.cpp]: argument list needs to be changed according to new indent policy

File: libsolidity/AST.h (hunk 1 above)
Comment (chriseth) [inline libsolidity/AST.h]: Unaligning everything according to coding style.

Comment (LianaHus) [inline libsolidity/AST.h]: I mean unalign others too :)

**Pull request merged**